injectproxy/routes.go

```diff
@@ -257,23 +258,40 @@ func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 }
 
 // getLabelValue returns the statically set label value, or the label value
-// sent through a URL parameter.
+// sent through a URL parameter or http request header.
 // It returns an error when either the value is found in both places, or is not found at all.
 func (r *routes) getLabelValue(req *http.Request) (string, error) {
+	headerValue, ok := req.Header[headerPrefix+r.label]
```

Comment:
To avoid having deep nesting, maybe we can call one function to get the label value. This function will first use the header, and if it's empty, it will fall back to the query parameter.

Then once we have a value, we can check if an explicit value through `-value` is set and return an error if that's true.

Reply:
:+1: